File: leetcode/problems/cpp/height-checker.cpp
// by @codeAbinash
// Time : O(n)
// Space : O(n)

#include "vector"
#include "algorithm"
using namespace std;

class Solution {
public:
  int heightChecker(vector<int>& heights) {
    vector<int>arr = heights;
    int ans = 0;
    sort(arr.begin(), arr.end());
    for (int i = 0;i < heights.size();i++)
      if (arr[i] != heights[i])
        ans++;
    return ans;
  }
};

File: leetcode/problems/java/height-checker.java
import java.util.Arrays;

class Solution {
    public int heightChecker(int[] heights) {
        int[] sorted = heights.clone();
        Arrays.sort(sorted);
        int count = 0;
        for (int i = 0; i < heights.length; i++) {
            if (heights[i] != sorted[i]) {
                count++;
            }
        }
        return count;
    }
}